src/handlers/delegatedStakingHandler.ts

```diff
@@ -0,0 +1,192 @@
+  hash: event.hash,
+  source: event.source_addr,
+  stakeRef: event.stake_ref,
+  globalSnapshotHash: event.global_snapshot_hash,
+  timestamp: event.created_at,
+});
+
+const delegateStakeWithdrawResponses = (txs) =>
+  txs.map(delegateStakeWithdrawResponse);
+
+const delegateStakeBalanceChangeResponse = (change) => ({
+  globalSnapshotOrdinal: change.global_snapshot_ordinal,
+  address: change.address,
+  nodeId: change.node_id,
+  balance: change.balance,
+  rewards: change.rewards,
+  timestamp: change.created_at,
+});
+
+const delegateStakeBalanceChangeResponses = (txs) =>
+  txs.map(delegateStakeBalanceChangeResponse);
+
+export const delegatedStakes = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        delegate_stake_withdraw_event: null,
+      },
+      orderBy: [{ ordinal: "desc" }],
+    },
+    prisma.delegate_stake_create_events.findMany,
+    delegateStakeCreateResponses
+  );
+};
+
+export const delegatedStake = async (event) => {
+  try {
+    const hash = event.pathParameters?.hash;
+
+    const stake = await prisma.delegate_stake_create_events.findUnique({
+      where: { hash },
+    });
+
+    return respond(stake, delegateStakeCreateResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const addressDelegatedStakes = async (event) => {
+  const address = event.pathParameters?.address;
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        source_addr: address,
+      },
+      orderBy: [{ ordinal: "desc" }],
+    },
+    prisma.delegate_stake_create_events.findMany,
+    delegateStakeCreateResponses
+  );
+};
+
+export const delegatedStakeWithdrawals = async (event) => {
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      orderBy: [{ created_at: "desc" }],
+    },
+    prisma.delegate_stake_withdraw_events.findMany,
+    delegateStakeWithdrawResponses
+  );
+};
+
+export const delegatedStakeWithdrawal = async (event) => {
+  try {
+    const hash = event.pathParameters?.hash;
+
+    const withdrawal = await prisma.delegate_stake_withdraw_events.findUnique({
+      where: { hash },
+    });
+
+    return respond(withdrawal, delegateStakeWithdrawResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const addressDelegatedStakeWithdrawals = async (event) => {
+  const address = event.pathParameters?.address;
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        source_addr: address,
+      },
+      orderBy: [{ created_at: "desc" }],
+    },
+    prisma.delegate_stake_withdraw_events.findMany,
+    delegateStakeWithdrawResponses
+  );
+};
+
+export const stakingBalanceByAddress = async (
```

Comment:
We may need to rethink this endpoint a bit. It shouldn't be a "balance change" endpoint. It should be a summary of Delegated Staking positions for users. It would be important to know the delegated stake hash and token lock hash along with the balances. 

Are we able to associate the create/update/withdraw records with these positions? We would ideally be able to return the current balance (even if withdrawn/zero) and all the associated records with that position along with a status (active, pendingWithdraw, withdrawn).